[SHOT-3906] Fixed menu favourites property.

rob-aitchison opened the pull request:
The favourites had an entry from the Project context so was not _None_ and thus would not re-build properly.
It should be safe to re-build this each time, Stacey? I don't see why not.
Thanks!

staceyoue commented on python/tk_vred/menu_generation.py:
```diff
@@ -39,9 +39,8 @@ def favourites(self):
-        if self._menu_favourites is None:
-            self._menu_favourites = self._engine.get_setting("menu_favourites", [])
-            self._menu_favourites.sort(key=lambda f: f["name"])
+        self._menu_favourites = self._engine.get_setting("menu_favourites", [])
+        self._menu_favourites.sort(key=lambda f: f["name"])
```
This is safe to rebuild each time, but I think this issue stems higher up from not clearing the menu favourites when we clear the menu when rebuilding the menu (e.g. on a context change).

We should instead just clear the menu favourites in `clean_menu` method
```
self._menu_favourites = None
```
and this should resolve the issue.

rob-aitchison replied:
Hey @staceyoue , I tried that and it does not seem to work.

staceyoue replied:
Ok, I think we should still set the menu favourites to None in the clean method regardless.

If this does not work, then I assume we are calling the `create_menu` method with `clean_menu=False`, which does not reset the menu favourites.

In this case, if we are rebuilding the menu favourites each time, we should change the `menu_favourites` property to a method:
```
def get_menu_favourites(self):
    // fetch the menu favourites from the settings
```

rob-aitchison replied:
@staceyoue my environment had an issue so my test above was invalid. :(

staceyoue reviewed (commented):
@rob-aitchison Did you forget to push a change to get the favourites?

rob-aitchison commented:
> @rob-aitchison Did you forget to push a change to get the favourites? or does it work now with just clearing the favourites in `clean_menu`?

Sorry @staceyoue , the latter now so I updated the comment and added a debug log message for good measure.

staceyoue reviewed (approved):
Nice!
